### ISSUE TITLE
Positional encodings do not support signal

### ISSUE THREAD

**Issue by haldenl** [vega/vega-lite#6434]
Here's an [example](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JACyUAVhDYA7EABoQAEzjQATjRyZ289AEEABBBoIcguIaZJ1h2DcyGA7nRiHETOMtXLDypJhUiioBKKigxEiCDGpoANqgYSD6wUxoAEwAHAC+ColoIABCqWgArKW5+egAwiWoYgDMlSBJACJ1AJwAjM1JAKJ1mT15LQUAYnWlTSNJAOJ1XR29BQASgwDsy+gAkpPp2QC6uSDI6gDWBdbqwXCyUGzKNLJkaKAAHq8gAGY0cILKBRQSkwAE8cHACmx1I9ZFFgkg3jQIJ9BEhPIJ9M8LGgAAzZY4gz4mMiwwQFeA0MhYED47JAA) where no marks are drawn as a result.

```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "A simple bar chart with embedded data.",
  "data": {
    "values": [
      {"a": "A", "b": 28},
      {"a": "B", "b": 55},
      {"a": "C", "b": 43},
      {"a": "D", "b": 91},
      {"a": "E", "b": 81},
      {"a": "F", "b": 53},
      {"a": "G", "b": 19},
      {"a": "H", "b": 87},
      {"a": "I", "b": 52}
    ]
  },
  "mark": "bar",
  "encoding": {
    "x": {"field": "a", "type": "ordinal", "axis": {"labelAngle": 0}},
    "y": {"signal": "height"}
  }
}
```

Looks like we are outputting

```
"y": {"signal": 20},
"y2": {"field": {"group": "height"}}
```

as opposed to

```
"y": {"signal": 20},
"y2": {"scale": "y", "value": 0}
```

**Comment by domoritz** [comment]
Note that the signal syntax is invalid as of https://github.com/vega/vega-lite/issues/6433. You need to write `"y": {"value": {"signal": "height"}}`

**Comment by haldenl** [comment]
will wait for #6433

**Comment by kanitw** [comment]
@domoritz @haldenl read my comments in #6433 too

**Comment by kanitw** [comment]
This will be fixed for free in #6438, which fixes #6433.

For example, this works:

```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "A simple bar chart with embedded data.",
  "data": {
    "values": [
      {"a": "A", "b": 28},
      {"a": "B", "b": 55},
      {"a": "C", "b": 43},
      {"a": "D", "b": 91},
      {"a": "E", "b": 81},
      {"a": "F", "b": 53},
      {"a": "G", "b": 19},
      {"a": "H", "b": 87},
      {"a": "I", "b": 52}
    ]
  },
  "mark": "bar",
  "encoding": {
    "x": {"field": "a", "type": "ordinal", "axis": {"labelAngle": 0}},
    "y": {"value": {"signal": "0"}}
  }
}
```

![image](https://user-images.githubusercontent.com/111269/80872026-462a0a80-8c64-11ea-93f6-4392f4261416.png)


Note that this is slight different from the example on the top:

1) We now wrap `signal` in `value`
2) The right value is 0, not height since y=height means the bottom of the chart.

3) We output `"y": {"signal": "0"}`, not something like

```
"y": {"signal": 20},
"y2": {"scale": "y", "value": 0}
```
(I don't know where this is really coming from).
